app/models/collection.rb

```diff
@@ -111,49 +105,49 @@
+                      with: /\A[A-Za-z0-9]\w*[A-Za-z0-9]\Z/ }
+  validates :icon_alt_text, length: { allow_blank: true, maximum: ArchiveConfig.ICON_ALT_MAX,
+                                      too_long: ts("must be less than %{max} characters long.", max: ArchiveConfig.ICON_ALT_MAX) }
+  validates :icon_comment_text, length: { allow_blank: true, maximum: ArchiveConfig.ICON_COMMENT_MAX,
+                                          too_long: ts("must be less than %{max} characters long.", max: ArchiveConfig.ICON_COMMENT_MAX) }
 
   validates :email, email_format: { allow_blank: true }
 
-  validates_presence_of :title, message: ts("Please enter a title to be displayed for your collection.")
-  validates_length_of :title,
-    minimum: ArchiveConfig.TITLE_MIN,
-    too_short: ts("must be at least %{min} characters long.", min: ArchiveConfig.TITLE_MIN)
-  validates_length_of :title,
-    maximum: ArchiveConfig.TITLE_MAX,
-    too_long: ts("must be less than %{max} characters long.", max: ArchiveConfig.TITLE_MAX)
+  validates :title, presence: { message: ts("Please enter a title to be displayed for your collection.") }
+  validates :title,
+            length: { minimum: ArchiveConfig.TITLE_MIN,
+                      too_short: ts("must be at least %{min} characters long.", min: ArchiveConfig.TITLE_MIN) }
+  validates :title,
+            length: { maximum: ArchiveConfig.TITLE_MAX,
+                      too_long: ts("must be less than %{max} characters long.", max: ArchiveConfig.TITLE_MAX) }
   validate :no_reserved_strings
   def no_reserved_strings
     errors.add(:title, ts("^Sorry, the ',' character cannot be in a collection Display Title.")) if
-      title.match(/\,/)
+      title.match(/,/)
   end
 
   # return title.html_safe to overcome escaping done by sanitiser
   def title
-    read_attribute(:title).try(:html_safe)
+    self[:title].try(:html_safe)
   end
 
-  validates_length_of :description,
-    allow_blank: true,
-    maximum: ArchiveConfig.SUMMARY_MAX,
-    too_long: ts("must be less than %{max} characters long.", max: ArchiveConfig.SUMMARY_MAX)
+  validates :description,
+            length: { allow_blank: true,
+                      maximum: ArchiveConfig.SUMMARY_MAX,
+                      too_long: ts("must be less than %{max} characters long.", max: ArchiveConfig.SUMMARY_MAX) }
 
-  validates_format_of :header_image_url, allow_blank: true, with: URI::regexp(%w(http https)), message: ts("is not a valid URL.")
-  validates_format_of :header_image_url, allow_blank: true, with: /\.(png|gif|jpg)$/, message: ts("can only point to a gif, jpg, or png file."), multiline: true
+  validates :header_image_url, format: { allow_blank: true, with: URI::DEFAULT_PARSER.make_regexp(%w[http https]), message: ts("is not a valid URL.") }
+  validates :header_image_url, format: { allow_blank: true, with: /\.(png|gif|jpg)$/, message: ts("can only point to a gif, jpg, or png file."), multiline: true }
```

Comment:
I don't understand what this is asking for...

Reply:
Pulled from a [different PR](https://github.com/otwcode/otwarchive/pull/4692) that also came across this warning:
> If you haven't had a chance to dig in, https://stackoverflow.com/a/577675 has some good information (I had it handy because we made a similar change in #4601).
> 
> TLDR - if someone changes the input to be multiline (like a textarea) they could, in theory enter something like
> 
> ```
> sneaky.jpeg
> haha, another line
> ```
> 
> IIRC in most languages, `$` and `\z` end up being functionally equivalent, but not in Ruby

So it's asking for the `with: /\.(png|gif|jpg)$/` to be changed to `with: /\.(png|gif|jpg)\z/`